src/CommunityToolkit/Xamarin.CommunityToolkit/Views/Popup/iOS/PopupRenderer.ios.cs

```diff
@@ -135,17 +141,26 @@ void SetViewController()
 		{
 			IVisualElementRenderer currentPageRenderer;
 			var modalStackCount = Application.Current.MainPage?.Navigation?.ModalStack?.Count ?? 0;
-			var mainPage = Application.Current.MainPage;
+			var page = Application.Current.MainPage;
 			if (modalStackCount > 0)
 			{
 				var index = modalStackCount - 1;
-				currentPageRenderer = Platform.GetRenderer(mainPage!.Navigation!.ModalStack![index]);
+				page = page!.Navigation!.ModalStack![index];
+				currentPageRenderer = Platform.GetRenderer(page);
+			}
+			else
+			{
+				currentPageRenderer = Platform.GetRenderer(page);
+			}
+
+			if (currentPageRenderer == null)
+			{
+				ViewController ??= page.CreateViewController();
 			}
 			else
 			{
-				currentPageRenderer = Platform.GetRenderer(mainPage);
+				ViewController ??= currentPageRenderer.ViewController;
```

Comment:
Are we sure that we just want to set the ViewController if it's null?

Reply:
previously, we always set it. it was done by CreateControl. it was the only 1 place where we set it.
now we also set ViewController in ctor. we do not want to set it here again.
hope I correctly understand your question.